modules/http/src/main/java/org/glassfish/grizzly/http/HttpServerFilter.java

```diff
@@ -794,7 +797,13 @@ protected void onHttpHeaderError(final HttpHeader httpHeader,
         final ServerHttpRequestImpl request = (ServerHttpRequestImpl) httpHeader;
         final HttpResponsePacket response = request.getResponse();
 
-        sendBadRequestResponse(ctx, response);
+        if (t instanceof URI.MalformedURIException) {
```

Comment:
If the exception knows the corresponding http status, this logic can be simplified